src/librustc_parse/parser/mod.rs

```diff
@@ -662,6 +662,21 @@ impl<'a> Parser<'a> {
                                 }
                             }
 
+                            // If this was a missing `@` in a binding pattern
+                            // bail with a suggestion
+                            // https://github.com/rust-lang/rust/issues/72373
+                            if self.prev_token.is_ident() && &self.token.kind == &token::DotDot {
+                                expect_err
+                                    .span_suggestion_short(
+                                        self.token.span,
+                                        &format!("maybe missing `@` in binding pattern?"),
+                                        "@..".to_string(),
```

Comment:
I think that the rustfmt style is for `ident @ ..`.

Reply:
updated the suggestion string to "@ .." in 0fdd31317a9c8d2fbc284546c711bc5d5139d83b

... but we don't get the space formatting relative to the `id` token that precedes the `@`.  

```
help: if you meant to bind the contents of the rest of the array pattern into `ts`, use `@`
   |
LL |         [h, ref ts@ ..] => foo(c, n - h) + foo(ts, n),
   |  
```

How do I increase the span back to the id token so that I can define the full suggestion string with `[id] @ ..` instead of `[id]@ ..`?